discovery-provider/src/tasks/entity_manager/entity_manager.py

```diff
@@ -183,6 +177,7 @@ def entity_manager_update(
                     original_records[record_type][entity_id].is_current = False
 
         # insert/update all tracks, playlist records in this block
+        session.flush()  # flush so aggregate triggers have invalidated records ^
```

Comment:
just to triple check my understanding of session.flush is that the whole tx can still be rolled back, this just communicates to the db that things are in the tx. it should be this way, but want to be absolutely 100% sure b/c the cost is painful as we've seen

Reply:
Yeah I added a test checking aggregate_playlist for example just to make sure!

You can see if I remove the session.flush the test will fail and produce these tables. playlists is correct with only 1 current repost but the aggregate table incorrectly shows 2 reposts because the trigger counted the original record before the invalidation was committed.
<img width="503" alt="Screen Shot 2022-09-13 at 2 22 07 PM" src="https://user-images.githubusercontent.com/6413636/190013407-9dc4d243-34c1-46d0-9180-aa98fdfe4086.png">
<img width="414" alt="Screen Shot 2022-09-13 at 2 21 58 PM" src="https://user-images.githubusercontent.com/6413636/190013414-2b92ca66-81ae-40b5-82cb-8208b74d71c8.png">